core/src/main/scala/tech/beshu/ror/accesscontrol/domain.scala

```diff
@@ -64,7 +65,7 @@ object domain {
   object User {
     final case class Id(value: NonEmptyString)
     object Id {
-      implicit val eqId: Eq[Id] = Eq.fromUniversalEquals
+      type UserIdCaseMappingEquality = CaseMappingEquality[User.Id]
```

Comment:
shouldn't we provide an Eq[User.Id] instance based on `UsernameCaseMapping`?

Reply:
`UsernameCaseMapping` is `Order`, so it's `Eq`